src/main/scala/viper/gobra/frontend/info/base/Type.scala

```diff
@@ -138,4 +140,18 @@ object Type {
       case t => Some(t)
     }
   }
+
+  /**
+    * Parameteric type for built-in members.
+    * `messages` maps from types of arguments to error messages.
+    * The partial function `typing` maps from types of arguments to the return type.
+    * It should be defined for all argument types for which no error message was returned by `messages`
+    */
+  case class AuxType(messages: (PExpression, Vector[Type]) => Messages, typing: Vector[Type] ==> FunctionT) extends Type
+
+  /**
+    * Parameteric type for built-in mpredicates and methods which are only parametric in their receiver type
+    * Applying `typing` to the receiver type results in a typle of types for the remaining arguments and the return type
+    */
+  case class SingleAuxType(messages: (PExpression, Type) => Messages, typing: Type ==> FunctionT) extends Type
```

Comment:
This is true. My idea was that `SingleAuxType` makes it explicit that there is only a dependency on the receiver type. If you use `AuxType` (or whatever other name we choose for it) you have to be careful to use the singleton vector containing the receiver type instead of the vector of argument types for methods and receivers.
However, I'm happy to unify this and just write this note as a comment. What do you prefer?

Reply:
I would prefer to unify it, but you can add a constructor for a single argument.